Gemfile

```diff
@@ -28,22 +28,14 @@ group :system_tests do
   gem 'beaker-puppet_install_helper', :require => false
 end
 
-is_x64 = Gem::Platform.local.cpu == 'x64'
-if is_x64
-  platform(:x64_mingw) do
-    gem "win32-dir", "~> 0.4.9", :require => false
-    gem "win32-process", "~> 0.7.4", :require => false
-    gem "win32-service", "~> 0.8.4", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-  end
-else
-  platform(:mingw) do
-    gem "win32-process", "~> 0.6.5", :require => false
-    gem "win32-service", "~> 0.7.2", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-    if RUBY_VERSION =~ /^1\./ then
-      gem "win32console", :require => false
-    end
+platforms :mswin, :mingw do
```

Comment:
Just as a quick follow up on this... 

I think we were able to determine that these gems are also listed here so that you can set `ENV['GEM_PUPPET_VERSION']` (see https://github.com/ferventcoder/puppetlabs-registry/blob/maint/master/gemfile-windows-relax-compatibility/Gemfile#L50-L56) to `file:///` so that you might test local unpublished Puppet changes in concert with changes to a gem.

Puppets `.gemspec` doesn't include any of the platform specific gems in https://github.com/puppetlabs/puppet/blob/master/.gemspec ... so without them in the `Gemfile`, there will be missing gems.

Supporting this workflow was key during some of the FFI development work IIRC.

Using 484cf18 (which is not the latest commit pushed anymore) and pointing to locally cloned Puppet source.

```
C:\source\puppetlabs-registry [pr/85]> ruby --version
ruby 2.1.5p273 (2014-11-13 revision 48405) [x64-mingw32]

C:\source\puppetlabs-registry [pr/85]> $ENV:GEM_PUPPET_VERSION = 'file://c:\source\puppet'

C:\source\puppetlabs-registry [pr/85]> bundle install --path .bundle\gems
DL is deprecated, please use Fiddle
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Installing CFPropertyList 2.3.1
Installing i18n 0.7.0
Installing rake 10.4.2
Installing minitest 5.7.0
Installing thread_safe 0.3.5
Installing addressable 2.3.8
Installing multi_json 1.11.1
Installing extlib 0.9.16
Installing mini_portile 0.6.2
Installing builder 3.2.2
Installing excon 0.45.3
Installing formatador 0.2.5
Installing json 1.8.3
Installing mime-types 2.6.1
Installing net-ssh 2.9.2
Installing ipaddress 0.8.0
Installing inflecto 0.0.2
Installing retriable 1.4.1
Installing multipart-post 2.0.0
Installing jwt 1.5.1
Installing open_uri_redirections 0.2.1
Installing hocon 0.9.2
Installing inifile 2.0.2
Installing trollop 2.1.2
Installing diff-lcs 1.2.5
Installing unf_ext 0.0.7.1
Installing json_pure 1.8.2
Installing metaclass 0.0.4
Installing rsync 1.0.9
Installing puppet-lint 1.1.0
Installing rspec-mocks 2.14.6
Using bundler 1.7.12
Installing rspec-core 2.14.8
Installing fission 0.5.0
Installing puppet-syntax 2.0.0
Installing launchy 2.4.3
Installing tzinfo 1.2.2
Installing autoparse 0.3.3
Installing docker-api 1.21.4
Installing nokogiri 1.6.6.2
Installing net-scp 1.2.1
Installing faraday 0.9.1
Installing rspec-expectations 2.14.5
Installing unf 0.1.4
Installing mocha 0.10.5
Installing activesupport 4.2.3
Installing rbvmomi 1.8.2
Installing fog-core 1.31.1
Installing aws-sdk-v1 1.64.0
Installing signet 0.6.1
Installing rspec 2.14.1
Installing fog-json 1.0.2
Installing fog-xml 0.1.2
Installing fog-local 0.2.1
Installing fog-vmfusion 0.1.0
Installing aws-sdk 1.64.0
Installing rspec-puppet 2.2.0
Installing google-api-client 0.8.2
Installing fog-sakuracloud 1.0.1
Installing fog-brightbox 0.7.2
Installing fog-serverlove 0.1.2
Installing fog-softlayer 0.4.7
Installing fog-atmos 0.1.0
Installing fog-storm_on_demand 0.1.1
Installing fog-ecloud 0.3.0
Installing fog-google 0.0.6
Installing fog-aws 0.6.0
Installing fog-powerdns 0.1.1
Installing fog-profitbricks 0.0.3
Installing fog-riakcs 0.1.0
Installing fog-radosgw 0.0.4
Installing fog-terremark 0.1.0
Installing fog-voxel 0.1.0
Installing puppetlabs_spec_helper 0.10.3
Installing fog 1.31.0
Installing beaker 2.15.1
Installing beaker-puppet_install_helper 0.2.1
Installing ffi 1.9.10
Installing win32-dir 0.4.9
Installing win32-security 0.2.5
Installing hiera 2.0.0
Installing facter 2.4.4
Using puppet 4.1.0 from source at c:/source/puppet
Your bundle is complete!
It was installed into ./.bundle/gems
Post-install message from nokogiri:
Nokogiri is built with the packaged libraries: libxml2-2.9.2, libxslt-1.1.28, zlib-1.2.8, libiconv-1.14.
```

This shows that `win32-dir` and `win32-security` get picked up (through `Facter 2.4.4` as a transitive dependency), but `win32-process` does not... (because `:mswin` is used as the gem platform)

Dependency management, don't you :heart: it?